catppuccin.tmux

```diff
@@ -31,6 +31,160 @@ setw() {
+
+  fi
+  
+  if [[ $fill == "number" ]] 
+  then
+    local show_number="#[fg=$background,bg=$color]$number"
+    local show_middle_separator="#[fg=$color,bg=$background,nobold,nounderscore,noitalics]$window_middle_separator"
+    local show_text="#[fg=$thm_fg,bg=$background]$text"
+
+    if [[ $window_number_position == "right" ]]
+    then
+      local show_left_separator="#[fg=$background,bg=$thm_bg,nobold,nounderscore,noitalics]$window_left_separator"
+      local show_right_separator="#[fg=$color,bg=$thm_bg]$window_right_separator"
+    fi
+
+    if [[ $window_number_position == "left" ]]
+    then
+      local show_right_separator="#[fg=$background,bg=$thm_bg,nobold,nounderscore,noitalics]$window_right_separator"
+      local show_left_separator="#[fg=$color,bg=$thm_bg]$window_left_separator"
+    fi
+
+  fi
+
+  local window_format
+
+  if [[ $window_number_position == "right" ]]
+  then
+    window_format="$show_left_separator$show_text$show_middle_separator$show_number$show_right_separator"
+  fi
+
+  if [[ $window_number_position == "left" ]]
+  then
+    window_format="$show_left_separator$show_number$show_middle_separator$show_text$show_right_separator"
+  fi
+
+  echo $window_format
+}
+
+build_status_module() {
+  local index="$1"
+  local icon="$2"
+  local color="$3"
+  local text="$4"
+
+  if [[ $index -eq 0 || $status_connect_separator == "no" ]]
+  then
+    local show_left_separator="#[fg=$color,bg=$thm_bg,nobold,nounderscore,noitalics]$status_left_separator"
+  else
+    local show_left_separator="#[fg=$color,bg=$thm_gray,nobold,nounderscore,noitalics]$status_left_separator"
+  fi
+
+  if [[ $status_fill == "icon" ]]
+  then
+    local show_icon="#[fg=$thm_bg,bg=$color,nobold,nounderscore,noitalics]$icon "
+    local show_text="#[fg=$thm_fg,bg=$thm_gray] $text"
+    local show_right_separator="#[fg=$thm_gray,bg=$thm_bg,nobold,nounderscore,noitalics]$status_right_separator"
+  fi
+
+  if [[ $status_fill == "all" ]]
+  then
+    local show_icon="#[fg=$thm_bg,bg=$color,nobold,nounderscore,noitalics]$icon "
+    local show_text="#[fg=$thm_bg,bg=$color]$text"
+    local show_right_separator="#[fg=$color,bg=$thm_bg,nobold,nounderscore,noitalics]$status_right_separator"
+  fi
+
+  if [[ $status_right_separator_inverse == "yes" ]]
+  then
+    local show_right_separator="#[fg=$thm_bg,bg=$color,nobold,nounderscore,noitalics]$status_right_separator"
+  fi
+
+  echo "$show_left_separator$show_icon$show_text$show_right_separator"
+}
+
+load_modules() {
+  local loaded_modules
+
+  local modules_path=$1
+  local modules_list=$2
+
+  local modules_array
+  read -a modules_array <<< "$modules_list"
+
+  local module_index=0;
+  local module_name
+  for module_name in ${modules_array[@]}
+  do
+    local module_path=$modules_path/$module_name.sh
+    source $module_path
+
+    if [[ 0 -eq $? ]]
+    then
+      loaded_modules="$loaded_modules$( show_$module_name $module_index )"
+      module_index=$module_index+1
```

Comment:
In portable shell, `$((` must be used for arithmetic, like this: `module_index=$((module_index + 1))`.